spark2/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -263,7 +263,7 @@ public String toString() {
     public DataWriter<InternalRow> createDataWriter(int partitionId, long taskId, long epochId) {
       Table table = tableBroadcast.value();
 
-      OutputFileFactory fileFactory = new OutputFileFactory(table, format, partitionId, taskId);
+      OutputFileFactory fileFactory = OutputFileFactory.builderFor(table, format, partitionId, taskId).build();
```

Comment:
Nit: Instead of keeping the constructor arguments, since we're using a builder, would it make sense to use a builder pattern constructor, such as the following?

Since these are all required fields, this might introduce more complexity than it is worth, but some of the fields _seem_ like things we might be able to determine other ways (such as partitionId). But I'm not quite sure if that's correct or not.

```
OutputFileFactor
    .builderFor(table)
    .withFormat(format)
    .withPartitonId(partitionId)
    .withTaskId(taskId)
    .build();
```

Reply:
I think the intent is to make table, partition, and task required. Otherwise, it isn't obvious that the builder will fail at runtime if you don't call `withTaskId`. It makes sense to me to avoid runtime issues by adding the required arguments to `builderFor` and anything that can be customized as a builder method.

Reply:
Correct, the intent was to indicate that these configs are required even in the API.